src/data/Dhis2TrackedEntityInstances.ts

```diff
@@ -46,9 +52,22 @@ export async function getTrackedEntityInstances(options: GetOptions): Promise<Tr
     for (const orgUnits of orgUnitsList) {
         // Limit response size by requesting paginated TEIs
         for (let page = 1; ; page++) {
-            const apiOptions = { api, program, orgUnits, page, pageSize };
+            const apiOptions = { api, program, orgUnits, page, pageSize, enrollmentStartDate, enrollmentEndDate };
             const { pager, trackedEntityInstances } = await getTeisFromApi(apiOptions);
-            apiTeis.push(...trackedEntityInstances);
+            const trackedEntityInstancesFilteredByEnrollmentIfSet =
+                enrollmentStartDate && enrollmentEndDate
+                    ? trackedEntityInstances.filter(trackedEntityInstance => {
+                          return trackedEntityInstance.enrollments.every(item =>
+                              moment(item.enrollmentDate).isBetween(
+                                  enrollmentStartDate,
+                                  enrollmentEndDate,
+                                  undefined,
+                                  "[]"
+                              )
+                          );
+                      })
+                    : trackedEntityInstances;
+            apiTeis.push(...trackedEntityInstancesFilteredByEnrollmentIfSet);
```

Comment:
@jvfdunkley Instead of manually filtering the TEIs, please use the API call.

In ``getTeisFromApi`` you can add two new optional params and use the ``programStartDate`` and ``programEndDate``

![image](https://user-images.githubusercontent.com/2181866/149734961-97ff7f31-9d66-4251-9f25-6f183c694dfd.png)